compiler/core/js_exp_make.ml

```diff
@@ -1604,6 +1605,12 @@ let int32_mul ?comment (e1 : J.expression) (e2 : J.expression) : J.expression =
 let unchecked_int32_mul ?comment e1 e2 : J.expression =
   {comment; expression_desc = Bin (Mul, e1, e2)}
 
+let int32_pow ?comment (e1 : t) (e2 : t) : J.expression =
+  match (e1.expression_desc, e2.expression_desc) with
+  | Number (Int {i = i1}), Number (Int {i = i2}) ->
+    int ?comment (Ext_int.int32_pow i1 i2)
```

Comment:
Have you checked that this is semantically equivalent to js power, or whatever `Pow` does below?

Reply:
This is different from js pow, especially when overflow. If I leave js pow out, wouldn't it cause undefined behavior in ReScript, like values ​​greater than `max_int`?

Reply:
Hmm, seems it is already possible by `n ** n`

Reply:
The intended question is: does the case with constants behave exactly like when 2 variables with the same constant values are passed?

Reply:
added more tests to verify it